app/services/playbook_downloader.rb

```diff
@@ -3,35 +3,35 @@
 # Service for dowloading SSG remediation playbooks
 class PlaybookDownloader
   class << self
-    def playbook_exists?(rule, profile = nil)
-      download(playbook_url(rule, profile&.short_ref_id)).present?
+    def playbooks_exist?(rules)
+      rules.inject({}) do |obj, rule|
+        obj.merge(rule.id => playbook_exists?(rule))
+      end
     end
 
-    def playbooks_exist?(rules)
-      rules.map do |rule|
-        [rule.id, playbook_exists?(rule)]
-      end.to_h
+    def playbook_exists?(rule)
+      playbook_list(rule.benchmark.os_major_version).include?(rule.short_ref_id)
     end
 
     private
 
-    def playbook_url(rule, profile_short_ref_id = nil)
+    def playbook_list(os_major_version)
+      @cache ||= {}
+      @cache[os_major_version] ||= begin
+        file = SafeDownloader.download(playbook_list_url(os_major_version))
+        JSON.parse(file.read).map do |row|
+          row['name'].sub(/\.yml$/, '')
+        end
+      end
+    end
+
+    def playbook_list_url(os_major_version)
       [
         Settings.compliance_ssg_url,
         'playbooks',
-        "rhel#{rule.benchmark.os_major_version}",
-        profile_short_ref_id || 'all',
-        "#{rule.short_ref_id}.yml"
+        "rhel#{os_major_version}",
+        'all'
```

Comment:
"all"? Where has the `profile_short_ref_id` gone?

Reply:
Where the `profile_short_ref_id` has been used? :shrug: :wink:

Reply:
I see. This is only used to update remediation_available boolean. 

Technically it could/should be profile specific. Could it happen that for some of the profiles the remediation would not be available?

Reply:
If that's even possible, our current DB structure has no means to store this kind of info and we'd need a remodel.

Reply:
@vkrizan I created a [task](https://issues.redhat.com/browse/RHICOMPL-2795) to verify it, but it should be independent from the current implementation as here I'm not changing behavior, just improving performance.